htlcswitch/switch.go

```diff
@@ -200,6 +201,13 @@ type Config struct {
 	// DustThreshold is the threshold in milli-satoshis after which we'll
 	// fail incoming or outgoing dust payments for a particular channel.
 	DustThreshold lnwire.MilliSatoshi
+
+	// SignAliasUpdate is used when sending FailureMessages backwards for
+	// option_scid_alias channels. This avoids a potential privacy leak by
+	// replacing the public, confirmed SCID with the alias in the
+	// ChannelUpdate.
+	SignAliasUpdate func(u *lnwire.ChannelUpdate) (*btcec.Signature,
```

Comment:
Why's this needed assuming a channel announcement+update already exist for a given failure update? Unclear why this new path is needed.

Reply:
Have to fail back with the scid used in the onion packet